std-bits/base/src/main/java/org/enso/base/enso_cloud/EnsoSecretHelper.java

```diff
@@ -22,6 +22,7 @@
 
 /** Makes HTTP requests with secrets in either header or query string. */
 public final class EnsoSecretHelper extends SecretValueResolver {
+  private static final EnsoHTTPResponseCache cache = new EnsoHTTPResponseCache();
```

Comment:
I think we should create an instance the first time `getCache` is called.

Creating it statically has some problems.

We check environment variables when instantiating `LRUCache` inside of `EnsoHTTPResponseCache` constructor.

If I recall correctly, in Native Image builds the static initialization is done at image build time and thus the environment variables read during static initialization would be the variables from the build machine, not the machine on which then we are being run. So deferring the initialization may prepare us better for Native Image.

Reply:
Additionally I saw throwing exceptions when the env variables are outside of ranges. Again, I don't think it's a good idea to do it inside of static initializer. Tbh I have no idea what will happen when we throw such an exception from a static initializer when we are running the GUI. I would have a guess that the Language Server will fail to boot in such a case. Which is not a good thing for users, especially as we are unlikely to have good reporting for such errors.

I'd move away to a 'lazy' initialization.

And overall I'm wondering if in this case we want throwing an exception. This will make the HTTP requests unusable due to misconfiguration. It feels like logging an error/warning and falling back to some defaults or clamping the values may be good enough - it will inform the user of the misconfiguration but will allow them to continue using the product.

Reply:
Done

Reply:
Reverting to default instead of throwing.